### ISSUE TITLE
Not receiving msg in real device 

### ISSUE THREAD

**Issue by imsidz** [shounakmulay/Telephony#164]
Everything is working fine in emulator but after build in release version installing any device its not receiving msg

**Comment by erndiggy** [comment]
Make sure to add appropriate permissions to the manifest file.

**Comment by msaadev** [comment]
I have this error too

`   <uses-permission android:name="android.permission.RECEIVE_SMS"/>
    <uses-permission android:name="android.permission.INTERNET"/>
    <uses-permission android:name="android.permission.READ_SMS"/>
    <uses-permission android:name="android.permission.READ_PHONE_STATE" />
    <uses-permission android:name="android.permission.READ_PHONE_NUMBERS"/>`

I have all permissions and receiver

**Comment by jk-aqreight** [comment]
anyone find the solution ?

**Comment by virtualars** [comment]
yes, with some devices it works, with others not, even samsung, not just xiaomi

**Comment by msaadev** [comment]
I write my codes in native side when message comes

**Comment by jk-aqreight** [comment]
@msaadev 
can you give us some example ?

**Comment by MohammadShateri** [comment]
This problem is very annoying.
The only problem is not working in the background.
Following all the points, but still there is this problem.
There is no solution for this problem?
This plugin is very useful but this is the problem I'm facing, I still haven't found a tool that supports in the background when sms is found.

**This plugin is no longer active, do you know a good alternative for it?**
**Or a solution that fixes the problem?**

**Comment by virtualars** [comment]
I had to change plugins, I found a very similar one, it doesn't ask for permissions automatically, so you need to install permission handler, but it works fine

**Comment by MohammadShateri** [comment]
@virtualars 
Does it really work on a real device? 
If it works on a real device, it will be great. Thank you for introducing this package to me to solve this problem. 🙏🌹

**Comment by virtualars** [comment]
> @virtualars Does it really work on a real device? If it works on a real device, it will be great. Thank you for introducing this package to me to solve this problem. 🙏🌹

https://pub.dev/packages/background_sms

**Comment by jk-aqreight** [comment]
> > @virtualars Does it really work on a real device? If it works on a real device, it will be great. Thank you for introducing this package to me to solve this problem. 🙏🌹
> 
> https://pub.dev/packages/background_sms

thanks. hope you can share us some sample code for using this package

**Comment by rahulmaindargi** [comment]
I am trying solution mentioned in https://github.com/shounakmulay/Telephony/issues/165

```
telephony:
  git:
    url: https://github.com/definev/Telephony.git
    ref: develop

```
change is adding `@pragma('vm:entry-point')` to `telephony.dart` method

Not sure if and when it will merged to original package.
